code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,507 @@
+		if("toggle_input")
+			turned_on = !turned_on
+			update_icon()
+		if("toggle_output")
+			firing = !firing
+			if(!firing)
+				destroy_lasers()
+			update_icon()
+
+		if("set_input")
+			input_number = clamp(params["set_input"], 0, 999) //multiplies our input by if input
+		if("set_output")
+			output_number = clamp(params["set_output"], 0, 999)
+
+		if("inputW")
+			power_format_multi = 1
+		if("inputKW")
+			power_format_multi = 1 KW
+		if("inputMW")
+			power_format_multi = 1 MW
+		if("inputGW")
+			power_format_multi = 1 GW
+
+		if("outputW")
+			power_format_multi_output = 1
+		if("outputKW")
+			power_format_multi_output = 1 KW
+		if("outputMW")
+			power_format_multi_output = 1 MW
+		if("outputGW")
+			power_format_multi_output = 1 GW
+
+/obj/machinery/power/transmission_laser/process()
+	max_grid_load = get_surplus()
+	input_available = get_surplus()
+	if(stat & BROKEN)
+		return
+
+	if(total_energy >= GLOB.announcement_threshold)
+		send_ptl_announcement()
+
+	var/last_disp = return_charge()
+	var/last_chrg = inputting
+	var/last_fire = firing
+
+	if(last_disp != return_charge() || last_chrg != inputting || last_fire != firing)
+		update_icon()
+
+	if(powernet && input_attempt && turned_on)
+		input_pulling = min(input_available, input_number * power_format_multi)
+
+		if(inputting)
+			if(input_pulling > 0)
+				consume_direct_power(input_pulling)
+				charge += input_pulling
+			else
+				inputting = FALSE
+		else
+			if(input_attempt && input_pulling > 0)
+				inputting = TRUE
+	else
+		inputting = FALSE
+
+	if(charge < MINIMUM_POWER)
+		firing = FALSE
+		output_level = 0
+		destroy_lasers()
+		return
+
+	if(!firing)
+		return
+
+	output_level = min(charge, output_number * power_format_multi_output)
+	if(!length(laser_effects))
+		find_blocker() // We aren't registering turf changes or atoms in the beam path while the beam is off.
+
+	if(blocker && QDELETED(blocker))// Checking here in case the blocker was destroyed by means other than the laser
+		find_blocker()
+
+	if(length(laser_effects))
+		for(var/obj/effect/transmission_beam as anything in laser_effects)
+			var/turf/beam_turf = get_turf(transmission_beam)
+			for(var/atom/beamed in beam_turf)
+				if(beamed)
+					beamed.ptl_beam_act(src)
+			// Make sure we beam the turf itself as well, in case it's a wall or something else that could be affected.
+			if(beam_turf)
+				beam_turf.ptl_beam_act(src)
+
+
+	if(!blocker)
+		sell_power(output_level * WATT_TICK_TO_JOULE)
+
+	charge -= output_level
+
+//// Selling defines are here
+#define MINIMUM_BAR 1
+#define PROCESS_CAP (20 - MINIMUM_BAR)
+
+#define A1_CURVE 20
+
+/obj/machinery/power/transmission_laser/proc/sell_power(joules)
+	var/mega_joules = joules / (1 MW)
+
+	var/generated_cash = (2 * mega_joules * PROCESS_CAP) / ((2 * mega_joules) + (PROCESS_CAP * A1_CURVE))
+	generated_cash += (4 * mega_joules * MINIMUM_BAR) / (4 * mega_joules + MINIMUM_BAR)
+	generated_cash = round(generated_cash)
+	if(generated_cash < 0)
+		return
```

Comment:
non-holding comment. I would definitely aim so that the average PTL setup (just using excess station power) generates around $2500 space credits an hour at most. This will double Engineering's income but also avoid just giving them a freebie for only setting up the PTL.

Reply:
The minimum is about 1 credit per cycle, which should be 1800 an hour.